src/Compiler/Service/FSharpProjectSnapshot.fs

```diff
@@ -668,11 +681,17 @@ and [<Experimental("This FCS API is experimental and subject to change.")>] FSha
         FSharpProjectSnapshot.FromOptions(options, getFileSnapshot)
 
 let rec internal snapshotToOptions (projectSnapshot: ProjectSnapshotBase<_>) =
+    let referencesOnDisk, otherOptions = // making sure we always produce the same order of flags and -r references
```

Comment:
This was an issue for the cache in the BackgroundCompiler as `AreSameForChecking` returned false when checking cache keys.